src/Sylius/Component/Core/OrderProcessing/StateResolver.php

```diff
@@ -38,22 +37,26 @@ public function resolvePaymentState(OrderInterface $order)
                 }
             }
 
-            if ($completedPaymentTotal >= $order->getTotal()) {
-                // Payment is completed if we have received full amount.
-                $paymentState = PaymentInterface::STATE_COMPLETED;
-            } else {
-                // Payment is processing if one of the payment is.
-                if ($payments->exists(function ($key, $payment) {
-                    return in_array($payment->getState(), [
-                        PaymentInterface::STATE_PROCESSING,
-                    ]);
-                })) {
-                    $paymentState = PaymentInterface::STATE_PROCESSING;
-                }
+            if (OrderInterface::STATE_CANCELLED === $order->getState()) {
+                $order->setPaymentState(OrderPaymentStates::STATE_CANCELLED);
```

Comment:
This should be a transition of state machine. We should never ever set states through setters.